### ISSUE TITLE
[TorrentLeech.pl] Results always discarded, because Radarr cannot handle foreign titles

### ISSUE THREAD

**Issue by chaostheory22** [Jackett/Jackett#8615]
**Jackett Version**: 0.16.324.0

### Description

I suspect Jackett was never able to find imdb id/link in releases from torrentleech.pl, but this wasn't a problem... until they changed naming of all releases from TorrentLeech.pl.

Example:

Bleed For This (2016)

Jackett finds the title: Opłacone krwią / Bleed.for.This.2016.MULTi.1080p.BluRay.x264.DTS.AC3-BiRD. 

But it's not mapping it correctly. The result is found, but radarr always says: "Failed to map movie, found title Opłacone krwią / Bleed for This, expected one of: Bleed for This, Paz, Paz: The Vinny Pazienza Story"

![image](https://user-images.githubusercontent.com/35978420/81736861-87f25680-9497-11ea-921d-1803bf5a53dc.png)

I suspect there's a problem with parsing the naming - Jackett/Radarr is not ignoring "/". I'm entirely aware that this could be radarr's fault, but Jackett could prevent that by finding imdb id/link which is always there - but for some reason not detected. 

![image](https://user-images.githubusercontent.com/35978420/81736253-8f653000-9496-11ea-8cdc-90aa32d60305.png)

While you're browsing manually there absolutely is an imdb image with embedded link to imdb page: 

![image](https://user-images.githubusercontent.com/35978420/81736311-a73cb400-9496-11ea-9efd-40012969e4b2.png)

I was trying these settings in Radarr's indexer settings: 

![image](https://user-images.githubusercontent.com/35978420/81736380-c20f2880-9496-11ea-9228-54cfe8eede6b.png)

But since the naming is double language - it's not working. I suspect improving Jackett to see imdb tags inside releases from torrentleech.pl will fix the situation.

Please don't crucify me, I tried every single config to make it work, mostly in Radarr indexer settings, but can't get it to work and therefore adding torrentleech.pl in Jackett is only viable while using manual search.

Best regards

**Comment by garfield69** [comment]
please do the following:
- enable jackett enhanced logging by ticking the **enhanced logging** checkbox and clicking the **apply server settings** button on your dashboard.
- on Radarr, disable temporarily all other indexers apart from the one you have for the Jackett TorrentLeech.pl indexer (by switching each Radarr indexer **enable rss** and **enable search** to **OFF**
- for the Radarr TorrentLeech.pl indexer, switch  **search by title** to **NO** to enable Radarr to send the imdbid to Jackett when it makes a query.
- search for the movie 
- fetch to Jackett log by following https://github.com/Jackett/Jackett/wiki/Troubleshooting#how-to-provide-a-jackett-enhanced-log-for-a-ticket

**Comment by chaostheory22** [comment]
I was searching for Big Miracle as an example:
![image](https://user-images.githubusercontent.com/35978420/81870443-1fc67200-9576-11ea-9dc4-2890a51ccee2.png)

[log.txt](https://github.com/Jackett/Jackett/files/4624961/log.txt)

**Comment by garfield69** [comment]
the log shows that in response to Radarr sending Jackett
```
http://192.168.1.11:9117/api/v2.0/indexers/torrentleech-pl/results/torznab/api?t=movie&cat=2000,2010,2020,2030,2035,2040,2045,2050,2060&extended=1&apikey=XXX&offset=0&limit=100&imdbid=1430615
```
Jackett sent the site
```
https://torrentleech.pl/browse.php?c65=1&c69=1&c66=1&c61=1&c67=1&c39=1&c36=1&c64=1&c59=1&c40=1&c68=1&search=tt1430615&incldead=1&titlesearch=1&polish=0&cat_film=&napisy=0
```
and the site returned one results
```
2020-05-13 23:56:31.0467 Info Found 1 (1 new) releases from Torrentleech.pl
```
which Jackett returned back to Radarr.
It is clear from your picture that Radarr got that result and chose to reject it
![image](https://user-images.githubusercontent.com/35978420/81870443-1fc67200-9576-11ea-9dc4-2890a51ccee2.png)

so
1. Jackett did an imdbid search with this indexer, and it works
2. the issue is that Radarr does not like the title it received despite that fact that it specifically asked for it by using the imdbid

it looks like the only way forward for this indexer to be any use with Radarr is to implement an option to strip the foreign language title at the start of the title string, everything up to the first forward slash.
I think we do something similar with some other indexers so I will do some research and see what I can come up with.

My biggest headache right now, is that for some reason the site is blocking my regular IP address, and rejecting IPs from my standby VPN hide.me
So I cannot login to my account and do testing of my code which sucks.

**Comment by garfield69** [comment]
>Jackett also isn't showing IMDB tag in manual search results.

your log shows that the HTML from the web site contains
```html
<tr class=darmowy>
<td align=center style='padding: 0px'><a href="browse.php?cat=39"><img border="0" src="https://torrentleech.pl/pic/fullhd.png" alt="Filmy/Full HD" /></a></td>
<td align=left><img src='pic/pl.jpg' /><img src='pic/napisy.gif' border=0 title='Napisy' alt='Napisy' /> <a href="details.php?id=237553&amp;hit=1">&nbsp;<b>Na ratunek wielorybom / Big.Miracle.2012.MULTi.1080p.BluRay.x264.DTS.AC3-BiRD</b></a><br><font size=1><font color=blue>(!!! PREMIERA FULL HD Z PL LEKTOREM !!! BiRD ZAPRASZA NA FILM !!!) (Melodramat) (Audio 5.1)</font>
</td>
<td align="center">0</td>
<td align=center><nobr>2020-05-12<br />11:24:54</nobr></td>
<td align=center>12.66<br>GB</td>
<td align=center>6 razy<br><a href="download.php/237553/Big.Miracle.2012.MULTi.1080p.BluRay.x264.DTS.AC3-BiRD.torrent?ssl=1"><b>[Pobierz]</b></a></td>
<td align=center><b><a href=details.php?id=237553&hit=1&toseeders=1><font color=#000000>5</font></a></b> /
0</td>
</table>
<p align="center"><b>1&nbsp;-&nbsp;1</b><br /><b>&lt;&lt;&nbsp;Poprzednia</b>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<b>Nast&#281;pna&nbsp;&gt;&gt;</b></p>
<script src="/anonymize.js" type="text/javascript"></script>
<script type="text/javascript"><!--
protected_links = "torrentleech.pl";

auto_anonymize();
//--></script></td></tr>
```
which does not contain an imdbid link, which is why the indexer which is looking for
```yaml
    imdb:
      optional: true
      selector: a[href*="www.imdb.com/title/tt"]
      attribute: href
```
cannot return anything to the Jackett results page.

there is no way for Jackett to present something that is not available in the HTML returned by the web site.

**Comment by chaostheory22** [comment]
OK, let me know if I can help you with anything.

**Comment by garfield69** [comment]
CinemaMovies is a Polish private site that has some titles in a similar layout to TorrentLeech-pl
So I was able to use the former site's indexer to develop the fix and test it, so I have confidence it should work for the latter indexer too.
The default is to leave the Polish title prefix in place, so after the update you will need to edit the indexer config and tick the **Drop the Polish title prefix** checkbox.
I'll post again when I publish the next Jackett release, which will include the enhancement.
Due in about 21 hours from now.

**Comment by garfield69** [comment]
Jackett 0.16.449

**Comment by chaostheory22** [comment]
@garfield69. I'm very sorry for tagging you, but I didn't know if this will pop up for you if I just responded to closed ticket. Your fix works almost flawlessly, but these guys from torrentleech are sloppy and therefore it's not always "Polish Title / English Title". I found that new releases with a typo in it like:
"Polish Title/English Title" (without spaces) or "Polish Title /English Title" (lack of space before english title". These are ignored by Jackett and "drop polish prefix" option won't work for them. If you find time to fix this, then I would really appreciate. Otherwise your fix works for 95% of releases anyways, so it's not that important, just saying. 

Thanks.
